src/main/java/neura/ui/Neura.java

```
@@ -181,14 +179,89 @@ private void addEventTask(String userInput) throws NeuraException{
-        System.out.println(SEPARATOR);
+    private void deleteTask(String userInput) throws NeuraException {
+        int taskIndex = getTaskIndex(userInput);
+        if (isValidTaskIndex(taskIndex)) {
+            Task removedTask = tasks.remove(taskIndex);
+            System.out.println(SEPARATOR);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(removedTask);
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+            System.out.println(SEPARATOR);
+            saveTasks();  // Save tasks to file after deletion
+        } else {
+            throw new NeuraException("Invalid task number.");
+        }
+    }
+
+    // Load tasks from file
+    private void loadTasks() {
+        File dataFile = new File(FILE_PATH);
+        if (dataFile.exists()) {
+            try (BufferedReader reader = new BufferedReader(new FileReader(dataFile))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    Task task = parseTask(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                }
+            } catch (IOException e) {
+                System.out.println("Error reading from file: " + e.getMessage());
+            }
+        }
+    }
+
+    // Parse a line into a Task object
+    private Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        try {
+            switch (parts[0]) {
+            case "T":
+                return new Todo(parts[2]);
+            case "D":
+                return new Deadline(parts[2], parts[3]);
+            case "E":
+                return new Event(parts[2], parts[3], parts[4]);
+            default:
+                return null;
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Error parsing line: " + line);
```

could consider throwing exception / logging error here instead of printing and returning null
or even introducing custom exceptions